### DISCUSSION TITLE
Dynamic Color for Bar ColumnGrid

### DISCUSSION THREAD

**Discussion by aldobc2** [visgl/deck.gl#8157] [Q&A]
Hello, I am using the ColumnGrid. Is there a way that depending on a certain value of my data, it's the color the that bar is going to be colored?

**Comment by Pessimistress** [comment]
Are you using ColumnLayer or GridLayer?

https://deck.gl/docs/api-reference/layers/column-layer#getfillcolor
https://deck.gl/docs/api-reference/aggregation-layers/grid-layer#coloraggregation

**Reply by aldobc2** [reply]
I am using ColumnLayer. I've already created a lambda function inside my get_fill_color accessor but it is filling all the objects with the same color. This is what I have: 

get_fill_color=[255,255 - ((lambda x: x*int((255/num)))("VOLUME")),0]

So you can have a bit more context, in my layer the elevation is defined by "VOLUME", so I want each column to be colored depending on the value of "VOLUME". So I thought of dividing 255 (max value for green) by num which is the max volume minus the min volume (did this because min doesn't start from 0). Then multiply the product by x which in this case would be my volume (this would give me values from 0 to 255) and subtracting that to 255 for coloring purposes. But it's not working!

**Reply by Pessimistress** [reply]
Oh you are using python not JavaScript. I am not familiar with the pydeck API, but I don't think you can define a lambda function like that. https://pydeck.gl/gallery/column_layer.html

**Reply by aldobc2** [reply]
Okok, and do you know by any chance if you can change the tooltip's information using the grid layer?
